### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mdanish09**

### CODE REVIEW

File: WeatherMan.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+#if the first Command line argument is "-a" this will execute
+if ARGV[0] == "-a"
+  year,month=ARGV[1].split('/')
+  dir = ARGV[2]
+  Weather_module.month_temperature(dir,year,month)
+end
+
+# if the first command line argument is "-e" this will execute
+if ARGV[0] == "-e"
```
Comment (devsinc) [inline WeatherMan.rb]: Use switch case instead of if else to compare these strings.

File: modules/weather_module.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
+require 'date'
+require './classes/weather.rb'
+
+class Weather_module
+  # MAp fro numbers to track month names 
+  @months=Hash.new("months")
```
Comment (devsinc) [inline modules/weather_module.rb]: why this line is required?

File: modules/weather_module.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      end
+      if value.mean_humidity.to_i>max_humid
+        max_humid=value.mean_humidity.to_i
+        max_humid_date=name
+      end
+    end
+    max_temperature[max_temp_date]=max_temp
+    min_temperature[min_temp_date]=min_temp
+    max_humidity[max_humid_date]=max_humid
+
```
Comment (devsinc) [inline modules/weather_module.rb]: a lot of rubocop offenses, make sure to configure rubocop and fix all offenses.

File: classes/weather.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      @max_visibility = max_visibility
+      @mean_visibility = mean_visibility
+      @min_visibility = min_visibility
+      @max_wind_speed = max_wind_speed
+      @mean_wind_speed = mean_wind_speed
+      @max_gust_speed = max_gust_speed
+      @precipitationmm = precipitationmm
+      @cloud_cover = cloud_cover
+      @events = events
+      @wind_dir_degrees = wind_dir_degrees
```
Comment (devsinc) [inline classes/weather.rb]: I don't see a lot of them being used in the code.
If an attr_accessor and instance variable is redundant then please remove that from the code.

**Review by devsinc — CHANGES_REQUESTED** [review]
A lot of rubocop offenses, kindly resolve them.

File: classes/weather.rb (hunk 4 above)
Comment (mdanish09) [reply classes/weather.rb]: It's just for the purpose of storing a line containing weather data of a single day
Without attribute accessor i am unable to access them when i make object of the class

**Pull request merged**